docs/serverless/AI-for-security/knowledge-base.asciidoc

```diff
@@ -0,0 +1,138 @@
+[[ai-assistant-knowledge-base]]
+= AI Assistant Knowledge Base
+
+AI Assistant's Knowledge Base feature enables AI Assistant to recall specific documents and other specified information. This information, which can include everything from the location of your datacenters to the latest threat research, provides additional context that can improve the quality of AI Assistant's responses to your queries. This topic describes how to enable and add information to Knowledge Base.
+
+NOTE: When you upgrade from {elastic-sec} version 8.15 to a newer version, information previously stored by AI Assistant will be lost. 
+
+.Requirements
+[sidebar]
+--
+
+* To use Knowledge Base, you need the `Elastic AI Assistant: All` privilege. To edit global Knowledge Base entries (information that will affect the AI Assistant experience for other users in the {kib} space), you need the `Allow Changes to Global Entries` privilege. 
+* You must <<security-ml-requirements, enable machine learning>> with a minimum ML node size of 4 GB.
+
+--
+
+[discrete]
+[[knowledge-base-rbac]]
+== Role-based access control (RBAC) for Knowledge Base
+
+The `Elastic AI Assistant: All` role privilege allows you to use AI Assistant and access its settings. It has two sub-privileges, `Field Selection and Anonymization`, which allows you to customize which alert fields are sent to AI Assistant and Attack Discovery, and `Knowledge Base`, which allows you to edit and create new Knowledge Base entries.
+
+image::images/knowledge-base-rbac.png[Knowledge base's RBAC settings,60%]
+
+[discrete]
+[[enable-knowledge-base]]
+== Enable Knowledge Base
+
+There are two ways to enable Knowledge Base.
+
+NOTE: You must individually enable Knowledge Base for each {kib} space where you want to use it.
+
+[discrete]
+=== Option 1: Enable Knowledge Base from an AI Assistant conversation
+
+Open a conversation with AI Assistant, select a large language model, then click **Setup Knowledge Base**. If the button doesn't appear, Knowledge Base is already enabled.
+
+image::images/knowledge-base-assistant-setup-button.png[An AI Assistant conversation showing the Setup Knowledge Base button]
+
+Knowledge base setup may take several minutes. It will continue in the background if you close the conversation. After setup is complete, you can access Knowledge Base settings from AI Assistant's conversation settings menu (access the conversation settings menu by clicking the three dots button next to the model selection dropdown).
```

Comment:
```suggestion
Knowledge Base setup may take several minutes. It will continue in the background if you close the conversation. After setup is complete, you can access Knowledge Base settings from AI Assistant's conversation settings menu (access the conversation settings menu by clicking the three dots button next to the model selection dropdown).
```